### PR TITLE
fix: detect if player gets stuck

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

A customer reported that their replay was getting stuck (see private thread: https://posthog.slack.com/archives/C0522G09STW/p1689355770362649)

On investigation there are some misconfigurations of their site (see private thread: https://posthog.slack.com/archives/C0460HY55M0/p1689356245975959)

When changing window their replay is getting stuck. Investigation showed the replayer timestamp would stop increasing at point of window change.

## Changes

It's hard to diagnose the source of this when the site is misconfigured, but this adds code to detect when the timestamp is stuck and skips forward "one frame"

## How did you test this code?

👀 locally

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 1]
```diff
@@ -172,6 +172,22 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
                 setCurrentTimestamp: (_, { timestamp }) => timestamp,
             },
         ],
+        timestampChangeTracking: [
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:175]: there are a bunch of spelling changes in this PR  to help reduce noise in my IDE 🙈 

But this is a real change, we already set the timestamp every time it changes so a reducer can track whether the timestamp has stopped changing

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 2]
```diff
@@ -622,13 +636,25 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
         updateAnimation: () => {
             // The main loop of the player. Called on each frame
             const rrwebPlayerTime = values.player?.replayer?.getCurrentTime()
-            let newTimestamp = values.fromRRwebPlayerTime(rrwebPlayerTime)
+            let newTimestamp = values.fromRRWebPlayerTime(rrwebPlayerTime)
+
+            const skip = values.playerSpeed * (1000 / 60) // rough animation fps
+            if (
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:642]: and then if the timestamp has got stuck we skip forward a frame

Comment (benjackwhite) [reply frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:642]: Does this work without calling `replayer?.play` and if so can we remove it?
The newTimestamp will trigger the codepaths that will eventually call `play` which feels safer as there are a lot of guards to ensure we don't play if the timestamp is outside the range of the current window for example

**Review by benjackwhite — CHANGES_REQUESTED** [review]
As someone who often merges 3 changes in one - I can't really complain about the rrweb upgrade 😉

**Comment by pauldambra** [comment]
> Does this work without calling replayer?.play and if so can we remove it?

@benjackwhite it doesn't... which is what makes me think this is an rrweb bug. It's like the player gets stuck.

**Review by benjackwhite — APPROVED** [review]
Urggg annoying but fair enough. Lets try it

**Pull request merged**